apps/web/pages/api/og.tsx

```diff
@@ -0,0 +1,291 @@
+import { ImageResponse } from '@vercel/og';
+import { NextRequest } from 'next/server';
+import { graphql, useLazyLoadQuery } from 'react-relay';
+import { fetchWithJustQueryText } from 'shared/relay/network';
+import styled from 'styled-components';
+
+import { TITLE_FONT_FAMILY } from '~/components/core/Text/Text';
+import { HEIGHT_OPENGRAPH_IMAGE, WIDTH_OPENGRAPH_IMAGE } from '~/constants/opengraph';
+import { ogPostQuery } from '~/generated/ogPostQuery.graphql';
+import colors from '~/shared/theme/colors';
+
+import { postIdQuery } from './postIdQuery';
+
+export const config = {
+  runtime: 'edge',
+};
+
+const ABCDiatypeRegular = fetch(
+  new URL('../../../../packages/shared/src/fonts/ABCDiatype-Regular.ttf', import.meta.url)
+).then((res) => res.arrayBuffer());
+
+const ABCDiatypeBold = fetch(
+  new URL('../../../../packages/shared/src/fonts/ABCDiatype-Bold.ttf', import.meta.url)
+).then((res) => res.arrayBuffer());
+
+const alpinaLight = fetch(
+  new URL('../../../../packages/shared/src/fonts/GT-Alpina-Standard-Light.otf', import.meta.url)
+).then((res) => res.arrayBuffer());
+
+type UrlSet = {
+  small: string | null;
+  medium: string | null;
+  large: string | null;
+};
+
+export default async function handler(request: NextRequest) {
```

Comment:
we should be able to stick this into `/api/og/post/[postId]`

and then grab the postId from `req.query` or `req.params` or `req.nextUrl`

this makes it so that when redirecting users here, we don't have to use:
```
gallery-og.vercel.app/api/og/post?postId=xxx
```

and instead
```
gallery-og.vercel.app/api/og/post/xxx
```

Reply:
`curl http://localhost:3001/api/og/post/12345` -> should test a very basic handler to ensure this works

file should be placed in `/api/og/post/[postId].tsx`

Reply:
collection handler would be placed in `/api/og/collection/[collectionId].tsx`